packages/js/product-editor/src/utils/product-apifetch-middleware.ts

```diff
@@ -4,6 +4,28 @@
 import apiFetch from '@wordpress/api-fetch';
 import { getQuery } from '@woocommerce/navigation';
 
+const routeMatchers = [
+	{
+		matcher: new RegExp( '^/wp/v2/product(?!_)' ),
+		replacement: '/wc/v3/products',
+	},
+	{
+		matcher: new RegExp( '^/wp/v2/product_variation' ),
+		replacement: '/wc/v3/products/0/variations',
+		getReplaceString: (): string => {
+			const query: { page?: string; path?: string } = getQuery();
+			const variationMatcher = new RegExp(
+				'/product/([0-9]+)/variation/([0-9]+)'
+			);
+			const matched = ( query.path || '' ).match( variationMatcher );
+			if ( matched && matched.length === 3 ) {
+				return '/wc/v3/products/' + matched[ 1 ] + '/variations';
+			}
+			return '/wc/v3/products/0/variations';
+		},
+	},
+];
```

Comment:
Why is this needed? Oh wait... I think the comment in the code below this answers it.

Reply:
Yeah, this is kinda dirty!
I was also floating the idea of always passing an additional param like `useWCAPI: boolean` that it would use that over checking if we are on the product editing page.
And in theory we want to push some changes up stream that allows us to use the `useEntity` store with our own endpoints ( or at-least register some new one's client side, so we don't run the risk of overwriting other ones, as WC Blocks relies on the `wp/v2` endpoint for example.